### ISSUE TITLE
Suggested keyword arguments break functionality

### ISSUE THREAD

**Issue by oliverklee** [bbatsov/rubocop#3675]
This happens with RuboCop 0.44.1. This issue started with version 0.44.0; version 0.43.0 was fine.

In our Rails 4.2 application, our controller specs use calls like these:

```ruby
    context 'with admin user logged in' do
      before { sign_in admin_user }

      it 'gets admin/organizations/edit with admin layout' do
        delete :destroy, id: organization

        assert_response :success

        expect(response).to render_template 'admin/organizations/destroy'
        expect(response).to render_template layout: 'layouts/admin'
      end
    end
```

RuboCop complains about this:

spec/controllers/admin/organizations_controller_spec.rb:200:9: C: Use keyword arguments instead of positional arguments for http call: delete.
        delete :destroy, id: organization
        ^^^^^^

When I change the code like this:

```ruby
delete :destroy, params: { id: organization }
```

… our specs (which have worked before) now fail:

Failures:

  1) Admin::OrganizationsController DELETE #destroy with admin user logged in gets admin/organizations/edit with admin layout
     Failure/Error: delete :destroy, params: { id: organization }
     
     ActionController::UrlGenerationError:
       No route matches {:action=>"destroy", :controller=>"admin/organizations", :params=>{:id=>"1"}}

So either the suggestion is not correct (at least not for Rails 4.2 - maybe it needs to be limited to Rails >= 5.0?), or I have just misunderstood the change suggested by RuboCop.

**Comment by Drenmi** [comment]
This cop only applies to Rails 5. How we're planning to deal with that in RuboCop is currently being discussed in #3629. 😄

**Comment by oliverklee** [comment]
Ah, thanks, good to know. Then I'll disable the cop until we migrate that project to Rails 5.

**Comment by oliverklee** [comment]
(closed as a duplicate of #3629)

**Comment by fengzhianshang** [comment]
`Rails/HttpPositionalArguments`
This cop is used to identify usages of http methods like get, post, put, patch without the usage of keyword arguments in your tests and change them to use keyword args. This cop only applies to Rails >= 5 . If you are not running Rails < 5 you should disable # the Rails/HttpPositionalArguments cop or set your TargetRailsVersion in your .rubocop.yml file to 4.0, etc.
But, set TargetRailsVersion to 4.0 is not effect and show the following message：
`Warning: unrecognized cop TargetRailsVersion found in /Users/gaojingdian/Workspace/Rails/test/.rubocop.yml`
finally, I disable this cop
